tokio/src/io/driver/scheduled_io.rs

```diff
@@ -503,12 +539,16 @@ cfg_io_readiness! {
                         let tick = TICK.unpack(scheduled_io.readiness.load(Acquire)) as u8;
 
                         // Safety: State::Done means it is no longer shared
+                        // TODO: I'm not sure why this safety claim is true. What about
+                        // it being `Done` prevents it from being shared? As far as I can tell
+                        // the `wake0` method in the scheduledio doesn't care about this being
+                        // Done?
```

Comment:
I agree it's a bit unclear, but I think it's correct. How can it transition to `Done` while being shared?

Reply:
Let me get back to you on that, I have to think through it